File: frontend/src/types/responseRecommendation.ts
/** AI response-recommendation types — field-level mirror of backend
 * AIResponseRecommendationRead.
 *
 * Advisory only end-to-end: every action stays a suggestion until human
 * approval — nothing here is ever executable. There is no
 * risk-score field: EventRisk.score stays the single official score.
 */

/** The six response actions. Display-only union: the UI maps
 * them to readable labels but never invents new action semantics. */
export type ResponseAction =
  | 'block_source_ip'
  | 'isolate_host'
  | 'disable_account'
  | 'hunt_related_activity'
  | 'escalate_to_incident'
  | 'monitor_only'

export interface RecommendationItem {
  action: ResponseAction
  /** Analyst-readable string — never an executable payload. */
  target: string
  rationale: string
}

export interface AIResponseRecommendation {
  id: string
  alert_group_id: string
  provider: string
  model: string
  overall_rationale: string
  recommendations: RecommendationItem[]
  confidence: number
  created_at: string
  updated_at: string
}

File: backend/app/schemas/response_recommendation.py
"""Pydantic schemas of the AI response-recommendation API."""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class RecommendationItemRead(BaseModel):
    """One recommended action as returned by the API.

Advisory only: action comes from the RESPONSE_ACTIONS vocabulary and
target is an analyst-facing string — never an executable payload.
"""

    action: str
    target: str
    rationale: str


class AIResponseRecommendationRead(BaseModel):
    """One AI response recommendation as returned by the API.

Mirrors the ai_response_recommendations row field-for-field; the output
protocol (overall_rationale / recommendations / confidence) surfaces
unchanged. No risk score is included — EventRisk.score stays the only
official score.
"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    alert_group_id: uuid.UUID
    provider: str
    model: str
    overall_rationale: str
    recommendations: list[RecommendationItemRead]
    confidence: float
    created_at: datetime
    updated_at: datetime
